### PR TITLE
vercel og image generation

### PR DESCRIPTION
**Pull request by Rohan-cp**
### Summary of Changes

Provide a brief description of the changes made in this PR. Explain the problem you are fixing or the feature you are adding.

### Demo or Before/After Pics

- If this is a new feature, include screenshots or recordings of the feature in action.
- If this PR makes visual changes, include before and after screenshots.

| Before                                     | After                                      |
| ------------------------------------------ | ------------------------------------------ |
|  |  |

### Edge Cases

List any common edge cases that you have considered and tested.

### Testing Steps

Provide steps on how the reviewer can test the changes.

### Checklist

Please make sure to review and check all of the following:

- [ ] I've tested the changes and all tests pass.
- [ ] (if web) I've tested the changes on various desktop screen sizes to ensure responsiveness.
- [ ] (if mobile) I've tested the changes on both light and dark modes.

### CODE REVIEW

File: apps/web/pages/api/og.tsx [hunk 1]
```diff
@@ -0,0 +1,291 @@
...
+    }
+  }
+
+  postImageUrl = postPreviewUrls?.large ?? '';
+
+  const ABCDiatypeRegularFontData = await ABCDiatypeRegular;
+  const ABCDiatypeBoldFontData = await ABCDiatypeBold;
+  const alpinaLightFontData = await alpinaLight;
+
+  return new ImageResponse(
```
Comment (Robinnnnn) [inline apps/web/pages/api/og.tsx:125]: lower priority - let's wrap this whole thing in a try/catch and in the catch return a new `ImageResponse` featuring the gallery logo as an SVG (this should be in our asset folders in the gallery repo)

we should also throw an error somewhere in here if the media isn't available. if so, `catch` will handle it and render the backup

File: apps/web/pages/api/og.tsx [hunk 2]
```diff
@@ -0,0 +1,291 @@
...
+  new URL('../../../../packages/shared/src/fonts/GT-Alpina-Standard-Light.otf', import.meta.url)
+).then((res) => res.arrayBuffer());
+
+type UrlSet = {
+  small: string | null;
+  medium: string | null;
+  large: string | null;
+};
+
+export default async function handler(request: NextRequest) {
```
Comment (Robinnnnn) [inline apps/web/pages/api/og.tsx:36]: we should be able to stick this into `/api/og/post/[postId]`

and then grab the postId from `req.query` or `req.params` or `req.nextUrl`

this makes it so that when redirecting users here, we don't have to use:
```
gallery-og.vercel.app/api/og/post?postId=xxx
```

and instead
```
gallery-og.vercel.app/api/og/post/xxx
```

Comment (Robinnnnn) [reply apps/web/pages/api/og.tsx:36]: `curl http://localhost:3001/api/og/post/12345` -> should test a very basic handler to ensure this works

file should be placed in `/api/og/post/[postId].tsx`

Comment (Robinnnnn) [reply apps/web/pages/api/og.tsx:36]: collection handler would be placed in `/api/og/collection/[collectionId].tsx`
